docker/Dockerfile.tests

```diff
@@ -29,3 +29,6 @@ RUN pip install --no-cache-dir --requirement requirements-dev-test.txt --constra
 
 FROM dev as dev-tools
 RUN pip install --no-cache-dir --requirement requirements-dev-tools.txt --constraint constraints-dev.txt
+
+# used to filter usage stats coming from CI and dev environments
```

Comment:
Let's move this to line 23, right after `FROM build_${SOURCE} AS dev`. This means the `dev` target and any derived targets (`test` and `dev-tools`) will have this set. The `test` target is what the azure pipelines use.

I've verified this works by building `test` after this change and seeing the env variable being set:
```
# echo ${GE_DATA_CONTEXT_ID}
00000000-0000-0000-0000-00000000e009
```

Reply:
Done